### PR TITLE
python3Packages.slackclient: 2.5.0 -> 2.9.3, python3Packages.slack-sdk: init at 3.3.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
- Update `slackclient` to latest upstream release 2.9.3. This is the last release of `slackclient`.
- The client was renamed to [`slack_sdk`](https://github.com/slackapi/python-slack-sdk). 
- Update of `errbot` to 6.1.7 (https://github.com/errbotio/errbot/blob/master/CHANGES.rst#v617-2020-12-18) to fix the build issue

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>errbot</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.slackclient</li>
    <li>python38Packages.slackclient</li>
    <li>python39Packages.slackclient</li>
  </ul>
</details>

File: pkgs/development/python-modules/slackclient/default.nix [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
   disabled = !isPy3k;
 
   src = fetchFromGitHub {
-    owner  = "slackapi";
-    repo   = pname;
-    rev    = version;
+    owner = "slackapi";
+    repo = pname;
+    rev = version;
     sha256 = "1ngj1mivbln19546195k400w9yaw69g0w6is7c75rqwyxr8wgzsk";
```
Comment (mweinelt) [inline pkgs/development/python-modules/slackclient/default.nix:30]: no hash update?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>errbot: log was empty</li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.slackclient</li>
<li>python38Packages.slackclient</li>
<li>python39Packages.slackclient</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111964` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.slackclient</li>
<li>python38Packages.slackclient</li>
<li>python39Packages.slackclient</li>
</ul>
</details>

File: pkgs/development/python-modules/slackclient/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/slackclient/default.nix]: Thanks. I didn't notice that `nix-update` failed in the first place.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.slack-sdk</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>errbot</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python37Packages.slack-sdk</li>
    <li>python37Packages.slackclient</li>
    <li>python38Packages.slack-sdk</li>
    <li>python38Packages.slackclient</li>
    <li>python39Packages.slackclient</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.slack-sdk</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>errbot</li>
    <li>python37Packages.slack-sdk</li>
    <li>python37Packages.slackclient</li>
    <li>python38Packages.slack-sdk</li>
    <li>python38Packages.slackclient</li>
    <li>python39Packages.slackclient</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.slack-sdk</li>
</ul>
</details>
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>errbot</li>
<li>python37Packages.slack-sdk</li>
<li>python37Packages.slackclient</li>
<li>python38Packages.slack-sdk</li>
<li>python38Packages.slackclient</li>
<li>python39Packages.slackclient</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111964` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.slack-sdk</li>
</ul>
</details>
<details>

  <summary>5 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.slack-sdk: log <a href="https://termbin.com/fgvn">https://termbin.com/fgvn</a></li>
<li>python37Packages.slackclient: log <a href="https://termbin.com/0qir">https://termbin.com/0qir</a></li>
<li>python38Packages.slack-sdk: log <a href="https://termbin.com/whda">https://termbin.com/whda</a></li>
<li>python38Packages.slackclient: log <a href="https://termbin.com/n2rd">https://termbin.com/n2rd</a></li>
<li>python39Packages.slackclient: log <a href="https://termbin.com/gdlp">https://termbin.com/gdlp</a></li>
  </ul>
</details>

File: pkgs/development/python-modules/slackclient/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/slackclient/default.nix]: Happens to me every once in a while as well, I'm not sure why that happens, but it should be investigated.

**Comment by mweinelt** [comment]
Broken on Python 3.9 due to uvloop.

```
error: Package ‘python3.9-uvloop-0.14.0’ in /home/hexa/git/nixos/master/pkgs/development/python-modules/uvloop/default.nix:71 is marked as broken, refusing to evaluate.
```

**Comment by fabaff** [comment]
> Broken on Python 3.9 due to uvloop.

There is no ETA for Python 3.9 support of uvloop (https://github.com/MagicStack/uvloop/issues/365).

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>errbot</li>
    <li>python37Packages.slack-sdk</li>
    <li>python37Packages.slackclient</li>
    <li>python38Packages.slack-sdk</li>
    <li>python38Packages.slackclient</li>
    <li>python39Packages.slack-sdk</li>
    <li>python39Packages.slackclient</li>
  </ul>
</details>

**Comment by flokli** [comment]
@fabaff is `slackclient` still being maintained? If not, we probably should introduce `slack-sdk` and remove `slackclient`.

**Comment by fabaff** [comment]
`slackclient` is still maintained according to an upstream statement and needed by Home Assistant and Errbot for now. I hope that
Home Assistant and Errbot will migrate to `slack-sdk` in the near future.

**Pull request merged**